### PR TITLE
Microsoft.Maui.Graphics

### PR DESCRIPTION
**Pull request by Clancey**
Updated everything to Microsoft.Maui.Graphics.

Compiled XAML still needs a little work.

### CODE REVIEW

File: src/Controls/samples/Controls.Sample.Droid/Controls.Sample.Droid.csproj [hunk 1]
```diff
@@ -79,10 +79,18 @@
...
+    </PackageReference>
+    <PackageReference Include="Xamarin.AndroidX.Lifecycle.LiveData">
+      <Version>2.3.0.1-$(_AndroidXVersion)</Version>
+    </PackageReference>
+    <PackageReference Include="Xamarin.Google.Android.Material">
+      <Version>1.3.0.1-$(_AndroidXVersion)</Version>
+    </PackageReference>
+    <PackageReference Include="Xamarin.AndroidX.Legacy.Support.V4">
+      <Version>1.0.0.7-$(_AndroidXVersion)</Version>
+    </PackageReference>
```
Comment (jonathanpeppers) [inline src/Controls/samples/Controls.Sample.Droid/Controls.Sample.Droid.csproj:93]: This doesn't seem right, did an IDE do this?

We should have all `<PackageReference/>` omit their versions so the versions are defined in a single place, such as:
* https://github.com/dotnet/maui/blob/main/eng/AndroidX.targets
* https://github.com/dotnet/maui/blob/main/eng/Microsoft.Extensions.targets

File: src/Core/src/Core-net6.csproj [hunk 2]
```diff
@@ -19,6 +19,7 @@
     <PackageReference Include="Microsoft.Extensions.Configuration" />
     <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" />
     <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" />
+		<PackageReference Include="Microsoft.Maui.Graphics" Version="6.0.100-preview.3.29" />
```
Comment (jonathanpeppers) [inline src/Core/src/Core-net6.csproj:22]: Can we use the same pattern here, too? Maybe an `eng/Microsoft.Maui.Graphics.targets` file with:
```
<PackageReference Update="Microsoft.Maui.Graphics" Version="6.0.100-preview.3.29" />
```
You would need to import the new `.targets` file here:
https://github.com/dotnet/maui/blob/e0b7d31cc600a9a2918489bb57cc2ae00c8e9d76/Directory.Build.targets#L4-L5

File: src/Core/src/Core.csproj [hunk 3]
```diff
@@ -10,8 +10,9 @@
 	</PropertyGroup>
 	<Import Project="..\..\..\.nuspec\Microsoft.Maui.Controls.MultiTargeting.targets" />
 	<ItemGroup>
-		<PackageReference Include="Microsoft.Extensions.Configuration" />
-		<PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" />
-		<PackageReference Include="Microsoft.Extensions.Logging.Abstractions" />
+		<PackageReference Include="Microsoft.Extensions.Configuration" Version="$(_MicrosoftHostingVersion)" />
+		<PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="$(_MicrosoftHostingVersion)" />
+		<PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="$(_MicrosoftHostingVersion)" />
+		<PackageReference Include="Microsoft.Maui.Graphics" Version="6.0.100-preview.3.29" />
```
Comment (jonathanpeppers) [inline src/Core/src/Core.csproj:16]: I would just undo this change.

File: src/Essentials/samples/Samples.Android/Samples.Android.csproj [hunk 4]
```diff
@@ -61,16 +61,20 @@
...
     <PackageReference Include="Xamarin.Forms.Visual.Material" Version="4.8.0.1821" />
     <PackageReference Include="System.Numerics.Vectors" Version="4.5.0" />
-    <PackageReference Include="Xamarin.Google.Android.Material" />
-    <PackageReference Include="Xamarin.AndroidX.AppCompat.AppCompatResources" />
+    <PackageReference Include="Xamarin.Google.Android.Material">
+      <Version>1.3.0.1-$(_AndroidXVersion)</Version>
+    </PackageReference>
+    <PackageReference Include="Xamarin.AndroidX.AppCompat.AppCompatResources">
+      <Version>1.2.0.7-$(_AndroidXVersion)</Version>
+    </PackageReference>
```
Comment (jonathanpeppers) [inline src/Essentials/samples/Samples.Android/Samples.Android.csproj:69]: Same here, shouldn't need this.

File: src/Essentials/test/DeviceTests.Android/Essentials.DeviceTests.Android.csproj [hunk 5]
```diff
@@ -60,9 +59,15 @@
...
-    <PackageReference Include="Xamarin.AndroidX.RecyclerView" />
+    <PackageReference Include="Xamarin.AndroidX.MediaRouter">
+      <Version>1.2.2.1-$(_AndroidXVersion)</Version>
+    </PackageReference>
+    <PackageReference Include="Xamarin.AndroidX.Palette">
+      <Version>1.0.0.7-$(_AndroidXVersion)</Version>
+    </PackageReference>
+    <PackageReference Include="Xamarin.AndroidX.RecyclerView">
+      <Version>1.1.0.8-net6preview01</Version>
+    </PackageReference>
```
Comment (jonathanpeppers) [inline src/Essentials/test/DeviceTests.Android/Essentials.DeviceTests.Android.csproj:70]: One more.

**Review by jonathanpeppers — COMMENTED** [review]
One question I have for when #603 is able to go in.

Is Microsoft.Maui.Graphics going to ship a `.nupkg` on nuget.org? Or is it supposed to *only* ship in the Maui workload?

When Maui is a workload, it either needs to distribute Microsoft.Maui.Graphics as a workload pack or put a `.nupkg` file in `dotnet/library-packs` so that offline restore can work.

File: src/Controls/samples/Controls.Sample.Droid/Controls.Sample.Droid.csproj (hunk 1 above)
Comment (Clancey) [reply src/Controls/samples/Controls.Sample.Droid/Controls.Sample.Droid.csproj:93]: Yes VSMAc did that, I didnt realized I missed a few

**Comment by hartez** [comment]
> Compiled XAML still needs a little work.

It would appear Clancey has a gift for understatement. 😀

**Comment by jsuarezruiz** [comment]
Some Brushes unit tests are failing, @hartez let me know if you want, I can take a look.

File: src/Compatibility/Maps/src/Android/Compatibility.Maps.Android.csproj [hunk 6]
```diff
@@ -17,6 +17,7 @@
     <ProjectReference Include="..\..\..\Core\src\Android\Compatibility.Android.csproj" />
   </ItemGroup>
   <ItemGroup>
+    <PackageReference Include="Microsoft.Maui.Graphics" Version="6.0.100-preview.3.34" />
```
Comment (jonathanpeppers) [inline src/Compatibility/Maps/src/Android/Compatibility.Maps.Android.csproj:20]: I still think the version from Microsoft.Maui.Graphics needs to be in a single place. It looks like it is in 6 places?

Comment (Redth) [reply src/Compatibility/Maps/src/Android/Compatibility.Maps.Android.csproj]: 🍰
